detekt/defs.bzl

```diff
@@ -127,6 +130,11 @@ detekt = rule(
             default = False,
             doc = "See [Detekt `--parallel` option](https://arturbosch.github.io/detekt/cli.html).",
         ),
+        "plugins": attr.label_list(
+            default = [],
+            doc = "[Detekt plugins](https://detekt.github.io/detekt/extensions.html). For example, [the formatting rule set](https://detekt.github.io/detekt/formatting.html). Labels should be JVM artifacts (generated via [`rules_jvm_external`](https://github.com/bazelbuild/rules_jvm_external) work).",
+            providers = [JavaInfo],
+        ),
```

Comment:
Seems like I’ve managed to get away without the toolchain. The Detekt Gradle plugin allows to declare plugins on the per-module (package, project) level so seems like we are on the right track.